src/main/scala/org/mauritania/main4ino/Server.scala

```diff
@@ -17,6 +17,10 @@ import org.mauritania.main4ino.security.AutherIO
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.FiniteDuration
 
+// If I replaced the below F=IO by some other Sync that does NOT delay the execution of side effects,
+// my program would stop having the property of referential transparency, correct? (side-effect expressions are evaluated
+// immediately without waiting for the end of the world).
+// What other Monad other than IO would it make sense to use here?
```

Comment:
1. Any worthy implementation of `Sync` should suspend execution of side effects.
2. This whole method could still be in `F`, and you could run the result of this method in `IO`. See https://github.com/pauljamescleary/scala-pet-store/blob/master/src/main/scala/io/github/pauljamescleary/petstore/Server.scala
3. Here you'd probably always use `IO`, but some of the instances for `IO` (e.g. `ApplicativeAsk`) could be provided by you manually.